src/main/cli.py

```diff
@@ -1,60 +1,115 @@
+PREFIXES = ["Add ", "Change ", "Create ", "Disable ", "Fix ",
+            "Merge ", "Move ", "Refactor ", "Release ",
+            "Remove ", "Rename ", "Tslint ", "Update "]
 
 DEFAULT = "\033[0;0m"
-VIOLET = DEFAULT+'\033[35m'
+WHITE = DEFAULT+'\033[37m'
+BLACK = DEFAULT+'\033[30m'
+RED = DEFAULT+'\033[31m'
 BLUE = DEFAULT+'\033[34m'
 CYAN = DEFAULT+'\033[36m'
 GREEN = DEFAULT+'\033[32m'
+VIOLET = DEFAULT+'\033[35m'
 YELLOW = DEFAULT+'\033[33m'
-RED = DEFAULT+'\033[31m'
 FILLER = '\033[;7m'
-WHITE = '\033[37m'
-BLACK = '\033[30m'
-BLACKFONE = FILLER+'\033[30m'
 WHITEFONE = FILLER+'\033[37m'
-GREENFONE = FILLER+'\033[32m'
+BLACKFONE = FILLER+'\033[30m'
+REDFONE = DEFAULT+'\033[31m'
 BLUEFONE = FILLER+'\033[34m'
+GREENFONE = FILLER+'\033[32m'
+YELLOWFONE = FILLER+'\033[33m'
 
 
-def show_template():
-    print(f"{GREEN}EXAMPLE: Refactor Z function in X file from Y component\n\
-         {BLUE}<optional part, adding it leave an empty line here>\n{GREEN}\
-         - Fix ...\n\
-         - Add ...\n\
-         - Remove ... \n{YELLOW}\n\
-HINT: to read chaos-hum team rules visit: {BLUE}\
-https://github.com/dimaka-wix/commit-msg-hook.git \n{DEFAULT}")
+def main(msg=None):
+    """
+    Entry point of commit-msg-hook.
+
+    If `msg` argument is None,
+    extract arguments from command line and run the hook logic
 
+    Args:
+        msg ([type], optional): Mainly used for testing in development.
+                                Defaults to None.
+    """
+    if msg is None:
+        msg = __get_commit_msg()
+        __set_args()
+    run_hook(msg)
 
-def main(argv=None):
-    if argv is None:
-        argv = __extract_msg()
-        __extract_args()
-    run_hook(argv)
 
+def run_hook(msg: str):
+    """
+    Run the main logic of the hook.
 
-def run_hook(msg):
-    __validate(msg)
+    If one of the checks failed abort commit(exit nonzero),
+    display appropriate errors and hints.
+    Otherwise exit zero and allow the `commit` command
+
+    Args:
+        msg (str): The commit message to check.
+    """
+    global IS_EXIT_1
     __check_length(msg)
     __check_subject_line(msg)
     __check_body(msg)
-    print(f"\n{GREEN}CONGRATS! Commit message matches the chaos-hub commit rules!\
-            \n{DEFAULT}")
+    if IS_EXIT_1:
+        show_example()
+        sys.exit(1)
+    else:
+        print(f"\n{GREEN}\
+CONGRATS! Commit message matches the chaos-hub commit rules 👍\n{DEFAULT}")
+        sys.exit(0)
+
+
+def show_example():
+    """
+    Display a commit message example.
+
+    The example follows chaos-hub team commit rules.
+    """
+    print(f"{GREEN}\
+EXAMPLE:\n\
+  Refactor foo function in ...\n{BLUE}\
+<body is optional, adding it leave an empty line here>\n{GREEN}\
+  - Fix ...\n\
+  - Add ...\n\
+  - Remove ... \n{YELLOW}\
+HINT:  to read chaos-hum team rules visit: {BLUE}\
+https://github.com/dimaka-wix/commit-msg-hook.git \n{DEFAULT}")
 
 
-def __extract_msg():
+def __get_commit_msg() -> str:
```

Comment:
Remove the private from all the functions and add above - __all__=["main","show_example"]